contracts/src/v0.8/automation/upkeeps/UpkeepBalanceMonitor.sol

```diff
@@ -0,0 +1,232 @@
+import {LinkTokenInterface} from "../../shared/interfaces/LinkTokenInterface.sol";
+import {Pausable} from "@openzeppelin/contracts/security/Pausable.sol";
+
+/// @title The UpkeepBalanceMonitor contract
+/// @notice A keeper-compatible contract that monitors and funds Chainlink Automation upkeeps.
+contract UpkeepBalanceMonitor is ConfirmedOwner, Pausable {
+  event ConfigSet(Config config);
+  event ForwarderSet(IAutomationForwarder forwarder);
+  event FundsWithdrawn(uint256 amountWithdrawn, address payee);
+  event TopUpFailed(uint256 indexed upkeepId);
+  event TopUpSucceeded(uint256 indexed upkeepId, uint96 amount);
+  event WatchListSet();
+
+  error InvalidConfig();
+  error InvalidTopUpData();
+  error OnlyForwarderOrOwner();
+
+  /// @member maxBatchSize is the maximum number of upkeeps to fund in a single transaction
+  /// @member minPercentage is the percentage of the upkeep's minBalance at which top-up occurs
+  /// @member targetPercentage is the percentage of the upkeep's minBalance to top-up to
+  /// @member maxTopUpAmount is the maximum amount of LINK to top-up an upkeep with
+  struct Config {
+    uint8 maxBatchSize;
+    uint24 minPercentage;
+    uint24 targetPercentage;
+    uint96 maxTopUpAmount;
+  }
+
+  // ================================================================
+  // |                           STORAGE                            |
+  // ================================================================
+
+  LinkTokenInterface private immutable LINK_TOKEN;
+
+  uint256[] private s_watchList;
+  Config private s_config;
+  IAutomationForwarder private s_forwarder;
+
+  // ================================================================
+  // |                         CONSTRUCTOR                          |
+  // ================================================================
+
+  /// @param linkToken the Link token address
+  /// @param config the initial config for the contract
+  constructor(LinkTokenInterface linkToken, Config memory config) ConfirmedOwner(msg.sender) {
+    require(address(linkToken) != address(0));
+    LINK_TOKEN = linkToken;
+    setConfig(config);
+  }
+
+  // ================================================================
+  // |                      CORE FUNCTIONALITY                      |
+  // ================================================================
+
+  /// @notice Gets a list of upkeeps that are underfunded
+  /// @return needsFunding list of underfunded upkeepIDs
+  /// @return topUpAmounts amount to top up each upkeep
+  function getUnderfundedUpkeeps() public view returns (uint256[] memory, uint256[] memory) {
+    uint256 numUpkeeps = s_watchList.length;
+    uint256[] memory needsFunding = new uint256[](numUpkeeps);
+    uint256[] memory topUpAmounts = new uint256[](numUpkeeps);
+    Config memory config = s_config;
+    IAutomationRegistryConsumer registry = getRegistry();
+    uint256 availableFunds = LINK_TOKEN.balanceOf(address(this));
```

Comment:
So this means this current contract holds LINK which will be used to auto fund the upkeeps. Is it possible to give a warning when this availableFunds is lower than topUpAmount as in L83?